Makefile

```diff
@@ -37,7 +37,11 @@ help: ## Display help
 
 presubmit: verify test ## Run all steps in the developer loop
 
-ci: toolchain verify licenses vulncheck battletest coverage ## Run all steps used by continuous integration
+ci-test: battletest coverage ## Runs tests and submits coverage
+
+ci-non-test: verify licenses vulncheck ## Runs checks other than tests
```

Comment:
Trying to think of more idiomatic names for this. I'd like to make sure we share as much as possible between karpenter and karpenter-core. 

I like this separation, since it enables developers to run `ci-test` locally without significant time investment of the license/vuln checks. I think we might be able to collapse `battletest` and `test`, since I'm not sure it actually takes very much more time 

`presubmit` Runs codegen, linting, testing, coverage, etc. It's fast 
`presubmit-slow` Runs license checks, vulnerabilities, slow stuff.  

This makes me think that we might actually want to run the slow stuff on postsubmit. These fail so rarely. It would be really nice to dramatically improve our merge times.

That would leave us with

`presubmit` Devs run before submission, Runs before CI checks. SLA <5min, ideally 3m?
`postsubmit` GH runs after submission, may take longer

Happy to chat offline

Reply:
Those are good idea, we also don't really need vulnerability scanning right here as we are doing it in another workflow. I'll do them in a follow up PR if you don't mind.